File: app/src/main/java/com/liuzhenli/reader/reciever/BatteryAndTimeChangeReceiver.java
package com.liuzhenli.reader.reciever;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.BatteryManager;
import android.text.TextUtils;

import com.micoredu.readerlib.helper.ReadConfigManager;
import com.micoredu.readerlib.page.PageLoader;

/**
 * Description:
 *
 * @author liuzhenli 2021/1/17
 * Email: 848808263@qq.com
 */
public class BatteryAndTimeChangeReceiver extends BroadcastReceiver {

    private final PageLoader mPageLoader;

    public BatteryAndTimeChangeReceiver(PageLoader pageLoader) {
        this.mPageLoader = pageLoader;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        if (ReadConfigManager.getInstance().getHideStatusBar()) {
            if (TextUtils.equals(Intent.ACTION_TIME_TICK, intent.getAction())) {
                if (mPageLoader != null) {
                    mPageLoader.updateTime();
                }
            }

            if (TextUtils.equals(intent.getAction(), Intent.ACTION_BATTERY_CHANGED)) {
                if (mPageLoader != null) {
                    mPageLoader.updateBattery(intent.getIntExtra(BatteryManager.EXTRA_LEVEL, 0));
                }
            }
        }
    }

    public void register(Context context, BatteryAndTimeChangeReceiver receiver) {
        IntentFilter filter = new IntentFilter();
        filter.addAction(Intent.ACTION_TIME_TICK);
        filter.addAction(Intent.ACTION_BATTERY_CHANGED);
        context.registerReceiver(receiver, filter);
    }

}

File: app-reader/src/main/java/com/micoredu/reader/reciever/BatteryAndTimeChangeReceiver.kt
package com.microedu.lib.reader.Reciever

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.BatteryManager
import android.content.IntentFilter
import com.micoredu.reader.constant.EventBus
import com.micoredu.reader.utils.postEvent

/**
 * Description:Battery and time receiver
 */
class BatteryAndTimeChangeReceiver : BroadcastReceiver() {
    val filter = IntentFilter().apply {
        addAction(Intent.ACTION_TIME_TICK)
        addAction(Intent.ACTION_BATTERY_CHANGED)
    }

    override fun onReceive(context: Context, intent: Intent) {
        when (intent.action) {
            Intent.ACTION_TIME_TICK -> {
                postEvent(EventBus.TIME_CHANGED, "")
            }
            Intent.ACTION_BATTERY_CHANGED -> {
                val level = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, -1)
                postEvent(EventBus.BATTERY_CHANGED, level)
            }
        }
    }
}